File: $_algotsu-kaizen/problemo-set/minimum-number-of-swaps-to-make-the-string-balanced.c

int minSwaps(char* s) {
    int depth = 0b1, length = 0, character = '\0';
    while (character = *(s + length++)) depth +=  character == '[' || depth == 0b1 ? 0b1 : -0b1;
    return depth / 0b10;
}

File: $_algotsu-kaizen/problemo-set/minimum-number-of-swaps-to-make-the-string-balanced.rs

impl Solution {

    pub fn min_swaps(s: String) -> i32 {
        let mut depth = 0b1;
        for character in s.chars() {
            depth = if (character == '[' || depth == 0b1) { depth + 0b1 } else { depth - 0b1 };
        }
        return depth / 0b10;
    }
}
